src/client/components/measures-tile/measures-tile.tsx

```diff
@@ -77,52 +76,42 @@ export class MeasuresTile extends React.Component<MeasuresTileProps, MeasuresTil
     });
   }
 
-  toggleMultiMeasure() {
+  toggleMultiMeasure = () => {
     var { clicker, essence } = this.props;
     clicker.toggleMultiMeasureMode();
     localStorage.set('is-multi-measure', !essence.getEffectiveMultiMeasureMode());
   }
 
-  render() {
-    var { essence, style } = this.props;
-    var { showSearch, searchText } = this.state;
-    var { dataCube } = essence;
-    var multiMeasureMode = essence.getEffectiveMultiMeasureMode();
-    var selectedMeasures = essence.getEffectiveSelectedMeasure();
-
-    var checkboxType: CheckboxType = multiMeasureMode ? 'check' : 'radio';
-
-    var shownMeasures = dataCube.measures.toArray();
-    if (searchText) {
-      shownMeasures = shownMeasures.filter((m) => {
-        return m.getTitleWithUnits().toLowerCase().indexOf(searchText.toLowerCase()) !== -1;
-      });
+  renderMessageIfNoMeasuresFound(measuresForView: MeasureOrGroupForView[]): JSX.Element {
+    const { searchText } = this.state;
+
+    if (!!searchText && !measuresForView.some(measure => measure.hasSearchText)) {
+      return <div className="message">{`No ${ STRINGS.measures.toLowerCase() } for "${searchText}"`}</div>;
+    } else {
+      return <></>;
```

Comment:
`null`, czy to znów kwestia TS? poza tym, nie potrzeba `else`, skoro w `if` jest `return`. I można z powodzeniem odwrócić warunek.

Reply:
Removing else will not work in TS but I will return null instead of empty fragment here